py/htm/encoders/eye.py

```diff
@@ -271,318 +410,69 @@ def input_space_sample_points(self, npoints):
-                corners = list(tuple(coord) for coord in np.transpose(corners))
-                # Draw the points
-                for start, end in zip(corners, corners[1:] + [corners[0]]):
-                    line_coords = (start[1], start[0], end[1], end[0],)
-                    draw.line(line_coords, fill='green', width=2)
-            del draw
-            plt.figure("Gaze Tracking")
-            im = np.array(im)
-            plt.imshow(im, interpolation='nearest')
-            plt.show()
-        return self.gaze[:]
-
-
-def small_random_movement(eye_sensor):
-    max_change_angle        = (2*3.14159) / 500
-    eye_sensor.position     = (
-        eye_sensor.position[0] + random.gauss(1, .75),
-        eye_sensor.position[1] + random.gauss(1, .75),)
-    eye_sensor.orientation += random.uniform(-max_change_angle, max_change_angle)
-    eye_sensor.scale = 1
 
+def _get_images(path):
+    """ Returns list of all image files found under the given file path. """
+    image_extensions = [
+        '.bmp',
+        '.dib',
+        '.png',
+        '.jpg',
+        '.jpeg',
+        '.jpe',
+        '.tif',
+        '.tiff',
+    ]
+    images = []
+    import os
+    if os.path.isfile(path):
+        basename, ext = os.path.splitext(path)
+        if ext.lower() in image_extensions:
+            images.append( path )
+    else:
+        for dirpath, dirnames, filenames in os.walk(path):
+            for fn in filenames:
+                basename, ext = os.path.splitext(fn)
+                if ext.lower() in image_extensions:
+                    images.append( os.path.join(dirpath, fn) )
+    return images
 
 if __name__ == '__main__':
-    eye = Eye()
-
-    import datasets, random
-    # data = datasets.Dataset('./datasets/small_items')
-    data = datasets.Dataset('./datasets/textures')
-    print("Num Images:", len(data))
-    data.shuffle()
-    for z in range(len(data)):
-        eye.reset()
-        data.next_image()
-        img_path = data.current_image
-        print("Loading image %s"%img_path)
-        img = np.asarray(PIL.Image.open(img_path))
-        eye.new_image(img)
-        eye.scale = 1
-
-        for i in range(10):
-            sdr = eye.compute()
-            eye.show_view()
-            small_random_movement(eye)
-
-    print("All images seen.")
+    import argparse
+    args = argparse.ArgumentParser()
+    args.add_argument('IMAGE', type=str)
+    args   = args.parse_args()
+    images = _get_images( args.IMAGE )
+    random.shuffle(images)
+    if not images:
+        print('No images found at file path "%s"!'%args.IMAGE)
+    else:
+        eye = Eye()
+        for img_path in images:
+            eye.reset()
+            print("Loading image %s"%img_path)
+            eye.new_image(img_path)
+            eye.scale = 1
+            for i in range(10):
+                pos,rot,sc = eye.small_random_movement()
+                sdr = eye.compute(pos,rot,sc) #TODO derive from Encoder
```

Comment:
Q: @ctrl-z-9000-times how should the encode look? 
- `encode(image, pos, rot, scale)`
- `encode(image, n_steps=10, saccadic_move_fn=small_random_moves)`

With related problem below, the output SDR is very dense. Would logical AND over the SDRs from saccadic steps describe the overall image?

Reply:
> encode(image, pos, rot, scale)

Including the eye's positioning in the arguments is a good idea.  One of the reasons I stopped working with vision is because I don't know how to make the HTM control where in the image the eye is looking,  So it makes sense for the user to control the eye positioning.

Reply:
> So it makes sense for the user to control the eye positioning. 

The question is the "level of management" we want to provide: 
- micro: `pos,rot,scale` represent positions in each of saccadic steps (ie. I'm looking at your left eye -> right eye -> mouth -> nose, ...) 
- macro: ditto represent "high level" position in the scene (ie I'm looking at "you"). 
  - and the encoder does (approximately) the saccadic movements (that is similar with our vision, we voluntarily control where we look at, but the micromanagement is subcontious). 

Intuitively, I'd be for the "macro" management mode, but how do we combine those sub-SDRs from saccades into one? logical-AND or OR?

Reply:
I think that this PR should do micro-level control over the eye movements.
1) Its what is already implemented
2) I don't really understand all of the science behind macro-level control

Reply:
`compute()` now expects coordinates for saccadic movements. 
Functionality is there, cannot derive from Encoder because #704